Some Docs suggestions (#114)
thautwarm/MLStyle.jl: docs/syntax/pattern.md

Pattern Matching
=======================

> Pattern mathing provide convenient ways to manipulate data. The basic syntax for pattern matching with MLStyle is of the form

Reviewer: ```suggestion
Pattern matching provides convenient ways to manipulate data. The basic syntax for pattern matching with MLStyle is of the form
```

```

Unlike most of ML languages or other libraries who only permit guards in the end of a case clause,
MLStyle.jl allows you to put guards anywhere during matching.

However, remember, due to some Julia optimization details, even if the guards can be put

> in the middle of a matching process, it is still better postpone it until the end of matching. This allows for better performance.

Reviewer: ```suggestion
in the middle of a matching process, it is still better to postpone it until the end of matching sequence. This allows for better performance.
```

true
```

> Destructuring Tuples, Arrays, and Dictionarys with Pattern Matching

Reviewer: ```suggestion
Destructuring Tuples, Arrays, and Dictionaries with Pattern Matching
```
